test/post_processor/test_post_processor.py

```diff
@@ -8,42 +8,157 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Tests for the MultiQubitPOVM class."""
+"""Tests for the POVMPostProcessor class."""
 
 from unittest import TestCase
 
-from povm_toolbox.library import ClassicalShadows
+import numpy as np
+from povm_toolbox.library import ClassicalShadows, LocallyBiasedClassicalShadows
 from povm_toolbox.post_processor import POVMPostProcessor
+from povm_toolbox.quantum_info.product_dual import ProductDUAL
 from povm_toolbox.sampler import POVMSampler
 from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
 from qiskit.primitives import StatevectorSampler as Sampler
 from qiskit.quantum_info import SparsePauliOp
 
 
 class TestPostProcessor(TestCase):
-    """TODO."""
+    """Test the methods and attributes of the :class:`.POVMPostProcessor class`."""
 
-    def test_init(self):
-        """TODO."""
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
 
         qc = QuantumCircuit(2)
         qc.h(0)
         qc.cx(0, 1)
 
-        sampler = Sampler()
+        sampler = Sampler(seed=1)
         povm_sampler = POVMSampler(sampler=sampler)
 
-        measurement = ClassicalShadows(num_qubits=2)
+        self.measurement = ClassicalShadows(num_qubits=2, seed_rng=2)
 
-        job = povm_sampler.run([qc], shots=256, povm=measurement)
+        job = povm_sampler.run([qc], shots=256, povm=self.measurement)
         result = job.result()
-        pub_result = result[0]
+        self.pub_result = result[0]
+
+    def test_init(self):
+        """Test that ``__init__`` works correctly."""
+        with self.subTest("Initialization with default ``dual``."):
+            post_processor = POVMPostProcessor(self.pub_result)
+            self.assertIsInstance(post_processor, POVMPostProcessor)
+            self.assertIsNone(post_processor._dual)
```

Comment:
Normally we would not test private attributes but here I agree with it because the `_dual` is the internal state of the `dual` property. And calling `post_processor.dual` would actually build a Dual out...